doc/user/http-frontend-ingest.rst

```diff
@@ -0,0 +1,358 @@
+
+- No backups are provided for tables ingested using this mechanism.
+- Binary data types are supported.
+
+  - **Binary data must be converted to an appropriate format for inclusion in the JSON request object.**
+  - See: :ref:`ingest-general-binary-encoding` (REST)
+
+- The API follows the same :ref:`http-frontend-general-error-reporting` scheme as adopted for :ref:`http-frontend-query`.
+
+Ingesting tables
+----------------
+
+The following REST service implements the synchronous interface for ingesting a table into Qserv:
+
+.. list-table::
+   :widths: 10 90
+   :header-rows: 0
+
+   * - ``POST``
+     - ``/ingest/data``
+
+A client must include a JSON object in the request body to specify the operation to be performed. The object follows this schema:
+
+.. code-block::
+
+    {   "database" :        <string>,
+        "table" :           <string>,
+        "binary_encoding" : <string>,
+        "timeout" :         <number>,
+        "schema" :          <array>,
+        "indexes" :         <array>,
+        "rows" :            <array>
+    }
+
+Where:
+
+``database`` : *string*
+  The required name of a user database. The names of the databases must start with the following prefix:
+
+  .. code-block::
+
+    user_<username>_
+
+  The rest of the name should include the name of a user or a role. For example:
+
+  .. code-block::
+
+    user_gapon
+
+``table`` : *string*
+  The required name of a table. Table names may not start with the following prefix:
+
+  .. code-block::
+
+    qserv_
+
+  This prefix is reserved for naming internal tables that Qserv places into user databases.
+
+``binary_encoding`` : *string* = ``hex``
+  The optional binary encoding of the binary data in the table. For further details see:
+
+  - :ref:`ingest-general-binary-encoding` (REST)
+
+``schema`` : *array*
+  The required schema definition. Each element of the array defines a column:
+
+  .. code-block::
+
+    {   "name" : <string>,
+        "type" : <string>
+    }
+
+  Where:
+
+  ``name``
+    The name of a column
+  ``type``
+    A valid MySQL type
+
+  **Note**: The service preserves the column order when creating a table.
+
+``indexes`` : *array* = ``[]``
+  The optional indexes will be created after ingesting the table.
+
+  More information on the index specification requirements can be found in the dedicated section of the document:
+  :ref:`http-frontend-ingest-indexes`.
+
+``rows`` : *array*
+  The required collection of the data rows to be ingested. Each element of the array represents a complete row,
+  where elements of the row correspond to the values of the respective columns:
+
+  .. code-block::
+
+    [ [ <string>, ... <string> ],
+                  ...
+      [ <string>, ... <string> ]
+    ]
+
+  **Note**:
+
+  - The number of elements in each row must match the number of columns defined in the table schema.
+  - The order of elements in each row should correspond to the order of columns in the table schema.
```

Comment:
Good question. In case of a mismatch,  the service will complain and refuse to execute the request. I have extended the documentation to mention this.